ugs-core/src/com/willwinder/universalgcodesender/AbstractController.java

```diff
@@ -729,11 +729,13 @@ public void commandSent(GcodeCommand command) {
     public void communicatorPausedOnError() {
         dispatchConsoleMessage(MessageType.INFO, "**** The communicator has been paused ****\n");
         try {
-            // FIXME the dispatchStateChange has logic for resuming send operations, this should have it's own handler.
-            // We can not use setCurrentState-method because in some cases the state change will not be dispatched
-            // if handleAllStateChangeEvents returns true.
-            dispatchStateChange(COMM_SENDING_PAUSED);
-            this.currentState = COMM_SENDING_PAUSED;
+            // Synchronize the controller <> communicator state.
+            if (!this.isStreaming()) {
+                this.comm.resumeSend();
+            }
+            else {
+                this.pauseStreaming();
```

Comment:
This will not dispatch a state change paused event if handleAllStateChangeEvents returns true. This would mean if we were running Check-mode and an error occurred the GUIBackend wouldn't get the paused event and activate the stop button. I must have missed a test for this.

To reproduce the problem run this in check mode and try to stop the file after the first error:
```gcode
G00 X22.375 Y54.9375
M3 S80
G01 X21.5625 Y54.9375
M5
G00 X21.125 Y54.9375
M3 S80
G01 X20.875 Y54.9375
M5
```

To make this work this must be added as well or if you come up with something better:
`dispatchStateChange(COMM_SENDING_PAUSED);`